### PR TITLE
Fixing Issue GAID fetch hang

### PR DESCRIPTION
**Pull request by sojanpr**
On Nexus 9 and Nexus 10 , some edge cases can lead to resource conflict
and dead lock while accessing GAID on background thread. Adding a timer
thread to access the GAID sms to solve this issue.
Issue is seen only with one specific partner and this solution seems
working fine with them.

@aaustin @Sarkar @EvangelosG

### CODE REVIEW

**Comment by aaustin** [comment]
looks good :+1:

File: Branch-SDK/src/io/branch/referral/ServerRequest.java [hunk 1]
```diff
@@ -262,25 +265,43 @@ private static ServerRequest getExtendedServerRequest(String requestPath, JSONOb
...
-            String advertisingId = sysObserver.getAdvertisingId();
-            if (advertisingId != null && getPost() != null) {
-                getPost().put(Defines.Jsonkey.GoogleAdvertisingID.getKey(), advertisingId);
-            }
-            int latVal = sysObserver.getLATValue();
-            if (getPost() != null) {
-                getPost().put(Defines.Jsonkey.LATVal.getKey(), latVal);
+    public void updateGAdsParams(final SystemObserver sysObserver) {
+        final CountDownLatch  latch  = new CountDownLatch(1);
+        skipOnTimeOut = false;
```
Comment (Sarkar) [inline Branch-SDK/src/io/branch/referral/ServerRequest.java]: hm - you have a skipOnTimeOut up there which is default set to false, do we need it again here?

**Comment by Sarkar** [comment]
:+1:

File: Branch-SDK/src/io/branch/referral/ServerRequest.java (hunk 1 above)
Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/ServerRequest.java]: This is to counter the case of calling "updateGAdsParams()" multiple time on same server request (Even though we  don't do it now)

**Pull request merged**